sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -48,6 +64,77 @@ export function isTracingDisabled(): boolean {
   return Boolean(azureTracingDisabledValue);
 }
 
+/**
+ * Maintains backwards compatibility with the previous `OperationTracingOptions` which included `SpanOptions`
+ */
+function disambiguateParameters<T extends { tracingOptions?: OperationTracingOptions }>(
+  operationOptions: T,
+  startSpanOptions?: SpanOptions
+): [OperationTracingOptions, SpanOptions, T] {
+  const { tracingOptions } = operationOptions;
+
+  // If startSpanOptions is provided, then we are using the new signature,
+  // otherwise try to pluck it out of the tracingOptions.
+  const spanOptions: SpanOptions = startSpanOptions || (tracingOptions as any)?.spanOptions || {};
+  spanOptions.kind = spanOptions.kind || SpanKind.INTERNAL;
+
+  return [tracingOptions || {}, spanOptions, operationOptions];
+}
+
+/**
+ * Creates a new span using the given parameters.
+ *
+ * @param spanName - The name of the span to created.
+ * @param spanOptions - Initialization options that can be used to customize the created span.
+ * @param tracingContext - The tracing context to use for the created span.
+ *
+ * @returns - A new span.
+ */
+function startSpan(spanName: string, spanOptions: SpanOptions, tracingContext: Context) {
+  if (isTracingDisabled()) {
+    return trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
+  }
+
+  const tracer = getTracer();
+  return tracer.startSpan(spanName, spanOptions, tracingContext);
```

Comment:
We could try to match the old style and send some logger.info messages suggesting to update. We can also wrap startSpan and catch errors, to include how to update to the latest version besides the internal errors

Reply:
This is all assuming @azure/core-tracing@next -> since older versions are already being used in the wild and we haven't seen any reports of OTel incompatibility with spanOptions.

Any version prior to 0.20.0 of OTel wouldn't work anyway because they won't have the APIs we need (specifically the APIs under the trace options which were moved into that namespace). So 0.18.0 for example would throw TypeError: api.trace.setSpan is not a function. 

But, testing with OTel 0.20.0 seems fine.

By the way, I learned recently that `yarn` is excellent for such experiments since you can essentially force a version of transitive dependencies using the `resolutions` field

```json
  "resolutions": {
    "@opentelemetry/api": "0.20.0"
  }
```

In summary things won't be any worse than they are today 😄

Reply:
ok ok ok!

Reply:
Quick update, I even tried something like this:

```ts
const { span, updatedOptions } = createSpan<any>("baz", {
  tracingOptions: { spanOptions: { foo: "bar", attributes: { "az.namespace": "baz" } } },
});
```

And I see that the unknown key foo was ignored in the span without a runtime error. 
Same when using the new param:

```ts
const { span, updatedOptions } = createSpan<any>("baz", {}, <any>{
  foo: "bar",
  attributes: { "az.namespace": "baz" },
});
```

So I think we're ok here personally